flake.nix

```diff
@@ -36,7 +36,10 @@
         { inherit (tests) vault-secrets; });
 
       legacyPackages = forSystems (system:
-        nixpkgs.legacyPackages.${system}.extend
-          self.overlays.default);
+        import nixpkgs {
+          inherit system;
+          config.allowUnfreePredicate = pkg: builtins.elem (pkg.pname) ["vault"];
```

Comment:
`allowUnfreePredicate` is a dangerous option, because it is not merged (nixpkgs does not use the module system here). That means, that if a user or other flake modules they use also defines a different `allowUnfreePredicate` one of them is silently overridden, afaik. I might be wrong on this, but at least it used to be that way and I've experienced it first-hand a while ago.

A workaround is to introduce a module:
```nix
{ config, lib, ... }:

{
  options = {
    allowUnfreePackages = lib.mkOption {
      default = [ ];
      type = lib.types.listOf lib.types.str;
      description = "List of regular expressions matching unfree packages allowed to be installed";
      example = lib.literalExpression ''[ "steam" "nvidia-.*" ]'';
    };
  };

  config = {
    nixpkgs.config.allowUnfreePredicate = pkg:
      let
        pkgName = (lib.getName pkg);
        matchPackges = (reg: ! builtins.isNull (builtins.match reg pkgName));
      in
      builtins.any matchPackges config.allowUnfreePackages;
  };
}
```
But that requires the user and all the flakes they use to be aware of it and never use `allowUnfreePedicate` directly.

Reply:
Yeah, I agree and AFAIK there is still no "official" convenient way to combine unfree predicates

Reply:
Perhaps, the better solution is to indeed avoid setting it on `vault-secrets` side and instead document that it should be used by the client to resolve nix evaluation errors

Reply:
Removed `allowUnfreePredicate` usage from modules and documented its usage instead